seismicpro/src/batch.py

```diff
@@ -300,3 +301,81 @@ def split_model_outputs(self, src, dst, shapes):
         else:
             raise ValueError(f"dst must be either `str` or `NamedExpression`, not {type(dst)}.")
         return self
+
+    @action
+    @inbatch_parallel(init='_init_component', target='for')
+    def crop(self, idx, src, origins, crop_shape, dst=None, joint=True, n_crops=1, n_overlaps=1, **kwargs):
+        """Crop batch components.
+
+        Parameters
+        ----------
+        src : str or list of str
+            Components to be cropped. Objects in each of them must implement `crop` method which will be called from
+            this method.
+        origins : list, tuple, np.ndarray or str
+            Origins define top-left corners for each crop or a rule used to calculate them. All array-like values are
+            cast to an `np.ndarray` and treated as origins directly, except for a 2-element tuple of `int`, which will
+            be treated as a single individual origin.
+            If `str`, represents a mode to calculate origins. Two options are supported:
+            - "random": calculate `n_crops` crops selected randomly using a uniform distribution over the source data,
+              so that no crop crosses data boundaries,
+            - "grid": calculate a deterministic uniform grid of origins, whose density is determined by `n_overlaps`.
+        crop_shape: tuple with 2 elements
```

Comment:
a lot of spaces are missing:(
```suggestion
        crop_shape : tuple with 2 elements
```

Reply:
Fixed everywhere